### DISCUSSION TITLE
为什么长轮询不直接返回变更数据

### DISCUSSION THREAD

**Discussion by xiaqi1210** [apolloconfig/apollo#4938] [Q&A] [answered]
为什么长轮询不直接返回变更数据, 而是返回notificationId, 再通过notificationId再调用一次拉取数据的接口去获取数据.
这样不就是增加了一次接口调用吗?
这样的方案带来的收益是什么?

**Comment by nobodyiam** [comment]
https://github.com/apolloconfig/apollo/issues/1887#issuecomment-455424350

**Reply by xiaqi1210** [reply]
我大概理解了, 其实就是解决并发写的时候推送到客户端的更新先后顺序问题是吧.
但实际上这个问题是直接返回变更数据的前提下也可以通过其他方案解决的吧.
我感觉主要还是为了设计上的简单是吗, 因为本身配置中心的请求量不大, 多一次请求不会造成什么压力

**Reply by nobodyiam** [reply, marked as answer]
保持逻辑简单确实是主要的一个考虑因素
